src/container/FieldBuilder/index.js

```diff
@@ -2,45 +2,59 @@ import classnames from 'classnames'
 import _ from 'lodash'
 import React from 'react'
 
-import Checkbox from '../../components/Checkbox'
 import Html from '../../components/Html'
 import Input from '../../components/Input'
 import Multiselect from '../../components/Multiselect'
-import Radio from '../../components/Radio'
 import Select from '../../components/Select'
+import SelectorList from '../../components/SelectorList'
 import Textarea from '../../components/Textarea'
 import { filteredKeys } from '../../utils/helpers'
-import {
-    getPlacement,
-    ifDefaultValue,
-    islabelHidden,
-} from '../../utils/inputSettings'
+import { ifDefaultValue, islabelHidden } from '../../utils/inputSettings'
 
 const FieldBuilder = ({ formData, presetValues = {}, register, errors }) => {
-    // The top level settings for the whole form
-    const formSettings = {
-        descriptionPlacement: formData.descriptionPlacement,
-    }
-
     // Loop through fields and create
     return formData.formFields.map(field => {
         // Set the wrapper classes
+        const {
+            descriptionPlacement: fieldDescPlace,
+            isRequired,
+            subLabelPlacement: fieldSubLabelPlace,
+            visibility,
+        } = field
+
+        const descriptionPlacement =
+            fieldDescPlace || formData.descriptionPlacement
+
+        const subLabelPlacement =
+            fieldSubLabelPlace || formData.subLabelPlacement
+
+        const fieldData = { ...field, descriptionPlacement }
         let inputWrapperClass = classnames(
+            'gfield',
             'gravityform__field',
             'gravityform__field__' + field.type,
             'gravityform__field--' + field.size,
             field.cssClass,
             { 'field-required': field.isRequired },
-            { 'hidden-label': islabelHidden(field.labelPlacement) }
+            { 'hidden-label': islabelHidden(field.labelPlacement) },
+            { gfield_contains_required: isRequired },
+            `field_sublabel_${subLabelPlacement}`,
```

Comment:
returning field_sublabel_null

Reply:
I will see if this is not getting the correct data or it just might not be set on your system and it doesn't really matter, we can always make them conditional

Reply:
Hmm, It's a little strange this was null for you, looking at my version ( think it's the latest) there are only two options in the config, but I just made it conditional so it hides it if no string comes through.